app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,99 @@
+require "pry-byebug"
+
+class InvitationsController < ApplicationController
+  def new
+    @cycle = Cycle.find(params[:cycle_id])
+    @invitation = Invitation.new
+    @invitation.cycle = @cycle
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params)
+    @cycle = Cycle.find(params[:cycle_id])
+    @invitation.cycle = @cycle
+    handle_email_name(@invitation.email)
+    if @invitation.save
+      respond_to do |format|
+        format.js
+      end
+    end
+  end
+
+  private
+
+  def invitation_params
+    params.require(:invitation).permit(:email, :message)
+  end
+
+  def handle_email_name(input)
```

Comment:
So these here could have been added to the model. Then you could use a before_validation to call these methods and set the name and email. Using a before validation callback would then allow us to populate the fields and then let the model validation the fields like they would have done anyway